### PR TITLE
Add unit tests to the JUnit xml report generation

### PR DESCRIPTION
**Pull request by romain-gilles-ultra**
Add unit tests to the JUnit XML report generation.

### CODE REVIEW

File: scalalib/src/mill/scalalib/TestModule.scala [hunk 1]
```diff
@@ -386,9 +387,11 @@ object TestModule {
                  tests={testResults.length.toString}
                  failures={testResults.count(_.status == Status.Failure.toString).toString}
                  errors={testResults.count(_.status == Status.Error.toString).toString}
-                 skipped={testResults.count(_.status == Status.Skipped.toString).toString}
+                 skipped={
+        testResults.count(testResult => SkippedStatus.contains(testResult.status)).toString
```
[the remark this answers is not in the document]
Comment (romain-gilles-ultra) [reply scalalib/src/mill/scalalib/TestModule.scala:391]: Hi @lefou 
You are right I should always use the `Enum.name()` function which is the right function to get the string representation of the enum name. I will fix it. BTW I introduced constants to for those name it's my mistake to not use them here ;)

File: scalalib/src/mill/scalalib/TestModule.scala [hunk 2]
```diff
@@ -206,6 +206,10 @@ trait TestModule
 
 object TestModule {
   private val FailedTestReportCount = 5
+  private val ErrorStatus = Status.Error.name()
+  private val FailureStatus = Status.Failure.name()
+  private val SkippedStatus =
```
Comment (lefou) [inline scalalib/src/mill/scalalib/TestModule.scala:211]:
```suggestion
  private val SkippedStates =
```

File: scalalib/src/mill/scalalib/TestModule.scala [hunk 3]
```diff
@@ -423,17 +431,17 @@ object TestModule {
         )
     ).getOrElse("")
     e.status match {
-      case Error if (e.exceptionMsg.isDefined && e.exceptionName.isDefined) =>
+      case ErrorStatus if (e.exceptionMsg.isDefined && e.exceptionName.isDefined) =>
```
Comment (lefou) [inline scalalib/src/mill/scalalib/TestModule.scala:434]: That's an interesting change. The old type must have been some kind of `Status` or `enum`, the new type is `String`. So either the old or the new version doesn't work or I'm missing something.

Comment (lefou) [reply scalalib/src/mill/scalalib/TestModule.scala:434]: Nevermind. Just realized, the renamed the `val`s.

**Review by lefou — APPROVED** [review]
Looks good to me. Thank you!

**Pull request merged**